django/thunderstore/repository/models/team.py

```diff
@@ -116,7 +117,12 @@ def validate(self):
 
     def save(self, *args, **kwargs):
         self.validate()
-        return super().save(*args, **kwargs)
+        add_namespace = False
```

Comment:
This logic really shouldn't be in the `save` method anyway. This needs to be executed within a transaction, which means you'll want to wrap a function such as `get_or_create_for_user`.

If we don't yet have a centralized function for creating teams, add one as a classmethod here, convert external Team creation calls to use that, and wrap it in a transaction where the namespace is set up correctly.

Reply:
Doned